### PR TITLE
Refactor/get rid of typeof ... "undefined"

### PR DESCRIPTION
**Pull request by jkoenig134**

### CODE REVIEW

**Comment by jkoenig134** [comment]
@coderabbitai review

**Comment by jkoenig134** [comment]
https://github.com/coderabbitai review

**Comment by jkoenig134** [comment]
@coderabbitai review

File: packages/transport/src/core/backbone/RESTClient.ts
Comment (britsta) [inline packages/transport/src/core/backbone/RESTClient.ts:1]: In line 373, there is the IF statement `if (typeof formData.getHeaders !== "undefined")`. As far as I know, this could probably be rephrased as well.

Also, in line 356, there is the logical statement `typeof Buffer !== "undefined"`. But that doesn't seem to be possible to reformulate. I also noticed something similar in `packages/transport/src/core/CoreError.ts` in line 61 with the IF statement `if (typeof Error.captureStackTrace !== "undefined")`.

File: .github/workflows/test.yml
Comment (Milena-Czierlinski) [inline .github/workflows/test.yml:1]: What does this do?

File: packages/transport/src/core/TransportContext.ts
Comment (Milena-Czierlinski) [inline packages/transport/src/core/TransportContext.ts:1]: Why was this deleted?

**Review by britsta — COMMENTED** [review]
General question: This PR is about getting rid of formulations like `typeof ... === "undefined"` or `typeof ... !== "undefined"`. What about formulations of the kind `... === undefined` or `... !== undefined` (for example the IF statement `if (address === undefined)` in line 31 of `packages/transport/src/core/types/CoreAddress.ts`), is an adjustment also wanted and possible for these?

File: .github/workflows/test.yml
Comment (Milena-Czierlinski) [reply .github/workflows/test.yml:1]: @sebbi08 just explained it to me

File: packages/transport/src/core/TransportContext.ts
Comment (jkoenig134) [reply packages/transport/src/core/TransportContext.ts:1]: it was bad and unused

**Comment by jkoenig134** [comment]
@britsta casting to boolean (for example `if (!"")`) with a nullish value would be false. So every time you handle optional strings (or more important booleans) you have to make sure be more accurate for js to understand what you want to do.

File: packages/transport/src/core/backbone/RESTClient.ts
Comment (jkoenig134) [reply packages/transport/src/core/backbone/RESTClient.ts:1]: I left out some things that make a difference in browsers. I will not touch that Buffer stuff. And for `Error.captureStackTrace` the compiler thinks that it is always defined.

File: .github/workflows/test.yml
Comment (jkoenig134) [reply .github/workflows/test.yml:1]: If you find an answer elsewhere - please write that down for everyone.

This stops pipeline runs that belong to the same PR so that there is always only one run per PR. Should help us with out GH Actions limits.

**Comment by Milena-Czierlinski** [comment]
Does this require a version bump for the small fixes?

**Comment by jkoenig134** [comment]
I don't made fixes, so no, no version bump required.

**Review by Milena-Czierlinski — APPROVED** [review]
